examples/specialty_plots/ishikawa.py

```diff
@@ -0,0 +1,232 @@
+            drawspines(data=data[k], ax=ax, parentspine=spine,
+                       alpha_ps=alpha_ps, alpha_ss=alpha_ss,
+                       spine_color=spine_color, reclevel=reclevel + 1,
+                       max_recursion_level=max_recursion_level)
+            # no primary_spines_rel_xpos is needed to be passed on recursion
+            # next spine settings - same level
+            alpha *= -1
+            spine_count += 1
+            if spine_count % 2 == 0:
+                s = s + spacing
+    return None
+
+
+def ishikawaplot(data, figsize=(20, 10), left_margin: float = 0.05,
+                 right_margin: float = 0.05, alpha_ps: float = 60.0,
+                 alpha_ss=0.0, primary_spines_rel_xpos: List[float] = [np.nan],
+                 pd_width: int = 0.1, spine_color: str = "black") -> plt.figure:
+    """
+
+    Parameters
+    ----------
+    data : TYPE
+        DESCRIPTION.
+    figsize : TYPE, optional
+        Matplotlib figure size. The default is (20,10).
+    left_margin : float, optional
+        Left spacing from frame border. The default is 0.05.
+    right_margin : float, optional
+        Right spacing from frame border. The default is 0.05.
+    alpha_ps : float, optional
+        First spine angle using during recursion. The default is 60.0.
+    alpha_ss : TYPE, optional
+        Secondary spine angle using during recursion. The default is 0.0.
+    primary_spines_rel_xpos : list, optional
+        Relative X-position of primary spines, a list of float is accepted.
+        The default is [np.nan].
+    pd_width : int, optional
+        Problem description box relative width. The default is 0.1.
+    spine_color : str, optional
+        Spine color. The default is "black".
+
+    Returns
+    -------
+    fig : matplotlib.pyplot.figure
+        Figure object containing the Ishikawa plot
+
+    """
+
+    fig = plt.figure(figsize=figsize)
+    ax = fig.gca()
+
+    # format axis
+    ax.set_xlim(0, 1.0)
+    ax.set_ylim(0, 1.0)
+    ax.axis('off')
+
+    # draw main spine
+    main_spine = ax.axhline(y=0.5, xmin=left_margin,
+                            xmax=1 - right_margin - pd_width,
+                            color=spine_color)
+
+    # draw fish head
+    props = dict(boxstyle='round', facecolor='wheat', alpha=1.0)
+    # add problem tag
+    ax.text((1 - right_margin - pd_width), 0.5, str.upper(list(data.keys())[0]),
+            fontsize=12, weight='bold',
+            verticalalignment='center', horizontalalignment='left', bbox=props)
+
+    # draw fish tail
+    x = (0.0, 0.0, left_margin)
+    y = (0.5 - left_margin, 0.5 + left_margin, 0.5)
+    ax.fill(x, y, color=spine_color)
+
+    # draw spines with recursion
+    drawspines(data=data[list(data.keys())[0]], ax=ax, parentspine=main_spine,
+               alpha_ps=alpha_ps, alpha_ss=alpha_ss,
+               primary_spines_rel_xpos=primary_spines_rel_xpos,
+               spine_color=spine_color)
+
+    plt.tight_layout()
```

Comment:
If you make an axes method, don't call tight_layout on the user.  If you keep `plt.figure` then use `plt.figure(figsize=figsize, layout='constrained')`.